tests/integration/cocoon/test_create_transaction_type.py

```diff
@@ -0,0 +1,107 @@
+import unittest
+from pathlib import Path
+import lusid
+import lusid.models as models
+from lusidtools.cocoon.utilities import create_scope_id
+from lusidtools.cocoon.transaction_type_upload import (
+    create_transaction_type_configuration,
+)
+import logging
+
+logger = logging.getLogger()
+
+
+class CocoonTestTransactionTypeUpload(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls) -> None:
+
+        secrets_file = Path(__file__).parent.parent.parent.joinpath("secrets.json")
+        cls.api_factory = lusid.utilities.ApiClientFactory(
+            api_secrets_filename=secrets_file
+        )
+
+        cls.new_alias = models.TransactionConfigurationTypeAlias(
+            type=create_scope_id().replace("-", ""),
+            description="TESTBUY1",
+            transaction_class="TESTBUY1",
+            transaction_group="SYSTEM1",
+            transaction_roles="AllRoles",
+        )
+
+        cls.movements = [
+            models.TransactionConfigurationMovementDataRequest(
+                movement_types="StockMovement",
+                side="Side1",
+                direction=1,
+                properties={},
+                mappings=[],
+            ),
+            models.TransactionConfigurationMovementDataRequest(
+                movement_types="CashCommitment",
+                side="Side2",
+                direction=-1,
+                properties={},
+                mappings=[],
+            ),
+        ]
+
+        cls.response = create_transaction_type_configuration(
+            cls.api_factory, cls.new_alias, cls.movements
+        )
+
+    def test_create_new_txn_type(self):
+
+        """
+        Create a new transaction type.
+        Verify that the new transaction type is created.
+        """
+
+        self.assertEqual(
+            set(
+                sorted(
+                    self.response.transaction_configs[-1].aliases[0].to_dict().items()
+                )
+            ),
+            set(self.new_alias.to_dict().items()),
+        )
+
+        print(self.response.transaction_configs[-1])
+
+        self.assertEqual(
+            sorted(
```

Comment:
I think your pattern of converting to sets and asserting on equality is better

Reply:
Agreed the set approach looks cleaner.
However I am not sure it will work when comparing two lists of movements.
The issue - I get the `TypeError: unhashable type` when trying to run set on list or dicts as lists/dicts are immutable.